cns/restserver/api.go

```diff
@@ -874,6 +862,40 @@ func (service *HTTPRestService) getNetworkContainerByID(w http.ResponseWriter, r
 	logger.Response(service.Name, reserveResp, resp.ReturnCode, err)
 }
 
+// the function is to get all network containers based on given OrchestratorContext
+func (service *HTTPRestService) getAllNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] getAllNetworkContainers")
+
+	var req cns.GetNetworkContainerRequest
+
+	logger.Printf("req is %+v", req)
+	err := service.Listener.Decode(w, r, &req)
+	logger.Request(service.Name, &req, err)
+	if err != nil {
+		logger.Errorf("[Azure CNS] failed to decode cns request due to %+v", err)
+		return
+	}
+
+	// getAllNetworkContainers gets called for multitenancy and
+	// setting the SDNRemoteArpMacAddress regKey is essential for the multitenancy
+	// to work correctly in case of windows platform. Return if there is an error
+	if err = platform.SetSdnRemoteArpMacAddress(); err != nil {
+		logger.Printf("[Azure CNS] SetSdnRemoteArpMacAddress failed with error: %s", err.Error())
+		return
+	}
+
+	getNetworkContainersResponse := service.getNetworkContainersResponse(req)
+	logger.Printf("getNetworkContainersResponse are %+v", getNetworkContainersResponse)
+	var resp cns.GetAllNetworkContainersResponse
+
+	resp.Response.ReturnCode = getNetworkContainersResponse[0].Response.ReturnCode
```

Comment:
we should not just copy the returnCode and Message from 0th NC. We should set those to indicate if all the NC responses succeeded or if it failed. Let's discuss this.